packages/instrumentation-socket.io/src/socket.io.ts

```diff
@@ -0,0 +1,217 @@
+            },
+            (moduleExports) => {
+                if (isWrapped(moduleExports.Server.prototype.emit)) {
+                    this._unwrap(moduleExports.Server.prototype, 'emit');
+                }
+                if (isWrapped(moduleExports.Server.prototype.on)) {
+                    this._unwrap(moduleExports.Server.prototype, 'on');
+                }
+            },
+            [broadcastOperatorInstrumentation, namespaceInstrumentation, socketInstrumentation]
+        );
+    }
+
+    private _patchOn() {
+        const self = this;
+        return (original: Function) => {
+            return function (ev: any, originalListener: Function) {
+                if (!self._config.traceReserved && reservedEvents.includes(ev)) {
+                    return original.apply(this, arguments);
+                }
+                const wrappedListener = function () {
+                    const operation = 'on';
+                    const messageName = ev;
+
+                    const span: Span = self.tracer.startSpan(`socket.io ${operation} ${messageName}`, {
+                        kind: SpanKind.CONSUMER,
+                        attributes: {
+                            [SemanticAttributes.MESSAGING_SYSTEM]: 'socket.io',
+                            [SemanticAttributes.MESSAGING_DESTINATION]: originalListener.name,
+                            [SemanticAttributes.MESSAGING_OPERATION]: MessagingOperationValues.PROCESS,
+                        },
+                    });
+
+                    if (self._config.onHook) {
+                        safeExecuteInTheMiddle(
+                            () => self._config.onHook(span, { args: arguments }),
+                            (e) => {
+                                if (e) diag.error(`socket.io instrumentation: onHook error`, e);
+                            },
+                            true
+                        );
+                    }
+
+                    const result = context.with(setSpan(context.active(), span), () =>
+                        originalListener.apply(this, arguments)
+                    );
+                    span.end();
+                    return result;
+                };
+
+                return original.apply(this, [ev, wrappedListener]);
+            };
+        };
+    }
+
+    private _patchEmit() {
+        const self = this;
+        return (original: Function) => {
+            return function (ev: any, ...args: any[]) {
+                if (!self._config.traceReserved && reservedEvents.includes(ev)) {
+                    return original.apply(this, arguments);
+                }
+                const operation = 'emit';
+                const messageName = ev;
+                const attributes = {
+                    [SemanticAttributes.MESSAGING_SYSTEM]: 'socket.io',
+                    [SemanticAttributes.MESSAGING_DESTINATION_KIND]: 'topic',
+                    [SemanticAttributes.MESSAGING_OPERATION]: operation,
```

Comment:
`operation` is not according to [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#operation-names):

<h3 style="box-sizing: border-box; margin-top: 24px; margin-bottom: 16px; font-size: 1.25em; font-weight: 600; line-height: 1.25; color: rgb(173, 186, 199); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(34, 39, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Operation names</h3><p style="box-sizing: border-box; margin-top: 0px; margin-bottom: 16px; color: rgb(173, 186, 199); font-family: -apple-system, system-ui, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(34, 39, 46); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">The following operations related to messages are defined for these semantic conventions:</p>
Operation name | Description
-- | --
send | A message is sent to a destination by a message producer/client.
receive | A message is received from a destination by a message consumer/server.
process | A message that was previously received from a destination is processed by a message consumer/server.